code/modules/mob/living/carbon/superior_animal/roach/types/toxic.dm

```diff
@@ -14,12 +14,40 @@
 	melee_damage_upper = 7 //Weaker than hunter
 	rarity_value = 22.5
 
+	ranged = 1
+	projectiletype = /obj/item/projectile/roach_spit
+	fire_verb = "spits glowing bile"
+	acceptableTargetDistance = 5
+	kept_distance = 3
+
 /mob/living/carbon/superior_animal/roach/toxic/UnarmedAttack(atom/A, var/proximity)
 	. = ..()
-	if(isliving(A))
-		var/mob/living/L = A
-		if(istype(L) && prob(25))
+	if(prob(25))
+		if(isliving(A))
+			var/mob/living/L = A
 			var/damage = rand(melee_damage_lower, melee_damage_upper)
-			L.damage_through_armor(damage, TOX)
+			L.apply_effect(40, IRRADIATE)
+			L.damage_through_armor(damage, TOX, attack_flag = ARMOR_BIO)
 			playsound(src, 'sound/voice/insect_battle_screeching.ogg', 30, 1, -3)
-			L.visible_message(SPAN_DANGER("\the [src] globs up some toxic bile all over \the [L]!"))
+			L.visible_message(SPAN_DANGER("\the [src] globs up some glowing bile all over \the [L]!"))
+
+/obj/item/projectile/roach_spit
+	name = "Glowing bile"
+	icon = 'icons/obj/hivemind.dmi'
+	icon_state = "goo_proj"
+	damage_types = list()
+	irradiate = 20
```

Comment:
Rad damage on basic mobs is a bad idea, it needs very special chemicals to cure.

Reply:
I survived it in testing, this is not just any roach but specifically a "radioactiveroach", hyronalin isn't all that special with a recipe of dylovene and radium, and once it's done processing all you need is dylovene to cure the Tox.